tapa/core.py

```diff
@@ -1513,6 +1509,80 @@ def get_aie_graph(self, task: Task) -> str:
             connect_def="\n\t\t".join(connect_def),
         )
 
+    def _visit_tasks(
```

Comment:
Could you add doc string for `_visit_tasks`? It has a general name and non-trivial logic.